[New] Show viewshed from geoelement in scene (#202)
Esri/arcgis-maps-sdk-swift-samples: Shared/Samples/Show viewshed from geoelement in scene/README.md

## How to use the sample

Tap to set a destination for the vehicle (a `GeoElement`). The vehicle will 'drive' towards the tapped location. The viewshed analysis will update as the vehicle moves.

## How it works

1. Create and show the scene, with an elevation source and a buildings layer.
2. Add a model (the `GeoElement`) to represent the observer (in this case, a tank).
    * Use a `SimpleRenderer` which has a heading expression set in the `GraphicsOverlay`. This way you can relate the viewshed's heading to the `GeoElement` object's heading.
3. Create a `GeoElementViewshed` with configuration for the viewshed analysis.
4. Add the viewshed to an `AnalysisOverlay` and add the overlay to the scene.
5. Configure the SceneView `CameraController` to orbit the vehicle.

## Relevant API

* AnalysisOverlay
* GeodeticDistanceResult
* GeoElementViewshed
* ModelSceneSymbol
* OrbitGeoElementCameraController

> * static GeometryEngine.geodeticDistance(from:to:distanceUnit:azimuthUnit:curveType:) (used to animate the vehicle)

Reviewer: ```suggestion
* static GeometryEngine.geodeticDistance(from:to:distanceUnit:azimuthUnit:curveType:)
```
Relevant API section should only contain symbols.

## About the data

> This sample shows a [Buildings in Brest, France Scene](https://www.arcgis.com/home/item.html?id=b343e14455fe45b98a2c20ebbceec0b0) from ArcGIS Online. The sample uses a [Tank model scene symbol](http://www.arcgis.com/home/item.html?id=07d62a792ab6496d9b772a24efea45d0) hosted as an item on ArcGIS Online.

Reviewer: Sorry I didn't make myself clear. Looks like other platforms are still using the South Africa scene, can you use that as well? Thanks.

Reviewer: Instead of using the portal item web scene, I'll update the common design to use the layer directly.

> This sample shows a [Buildings in Brest, France Scene](https://www.arcgis.com/home/item.html?id=b343e14455fe45b98a2c20ebbceec0b0) from ArcGIS Online. The sample uses a [Tank model scene symbol](http://www.arcgis.com/home/item.html?id=07d62a792ab6496d9b772a24efea45d0) hosted as an item on ArcGIS Online.

Reviewer 2: ```suggestion
This sample shows  [buildings in Brest, France Scene](https://www.arcgis.com/home/item.html?id=b343e14455fe45b98a2c20ebbceec0b0) from ArcGIS Online. The sample uses a [Tank model scene symbol](http://www.arcgis.com/home/item.html?id=07d62a792ab6496d9b772a24efea45d0) hosted as an item on ArcGIS Online.
```

# Show viewshed from geoelement in scene

> Analyze the viewshed for an object (GeoElement) in a scene.

Reviewer: Instead of using the API name in description, let's use the plain English from the [terminology guide](https://csreview.esri.com/terminology-guide)
```suggestion
Analyze the viewshed for an object (geoelement) in a scene.
```

## Use case

A viewshed analysis is a type of visual analysis you can perform on a scene. The viewshed aims to answer the question 'What can I see from a given location?'. The output is an overlay with two different colors - one representing the visible areas (green) and the other representing the obstructed areas (red).

## How to use the sample

Tap to set a destination for the vehicle (a GeoElement). The vehicle will 'drive' towards the tapped location. The viewshed analysis will update as the vehicle moves.

## How it works

1. Create and show the scene, with an elevation source and a buildings layer.
2. Add a model (the `GeoElement`) to represent the observer (in this case, a tank).
    * Use a `SimpleRenderer` which has a heading expression set in the `GraphicsOverlay`. This way you can relate the viewshed's heading to the `GeoElement` object's heading.
3. Create a `GeoElementViewshed` with configuration for the viewshed analysis.
4. Add the viewshed to an `AnalysisOverlay` and add the overlay to the scene.
5. Configure the SceneView `CameraController` to orbit the vehicle.

## Relevant API

> * AnalysisOverlay
> * GeodeticDistanceResult
> * GeoElementViewshed
> * GeometryEngine.DistanceGeodetic (used to animate the vehicle)
> * ModelSceneSymbol
> * OrbitGeoElementCameraController

Reviewer 2: I can't remember, should these use code voice? Also, `GeometryEngine.DistanceGeodetic` isn't a symbol in the Swift API.

```suggestion
* `AnalysisOverlay`
* `GeodeticDistanceResult`
* `GeoElementViewshed`
* `static GeometryEngine.geodeticDistance(from:to:distanceUnit:azimuthUnit:curveType:)` (used to animate the vehicle)
* `ModelSceneSymbol`
* `OrbitGeoElementCameraController`
```

Reviewer: No they don't need the backticks. Qualified symbols should be used

## Offline data

Offline sample data will be downloaded by the sample viewer automatically.

Link | Local Location
---------|-------|

> |[Model Marker Symbol Data](https://www.arcgis.com/home/item.html?id=07d62a792ab6496d9b772a24efea45d0)| `<userhome>`/ArcGIS/Runtime/Data/3D/bradley_low_3ds/bradle.3ds |

Reviewer 2: The local location is not correct. In fact, there is no local location for on-demand data used by this app.
